src/ansys/fluent/core/streaming_services/datamodel_event_streaming.py

```diff
@@ -53,30 +53,30 @@ def _process_streaming(self, id, stream_begin_method, started_evt, *args, **kwar
                     )
                 with self._lock:
                     self._streaming = True
-                    for tag, cb in self._cbs.items():
-                        if tag == response.tag:
-                            if response.HasField("createdEventResponse"):
-                                childtype = response.createdEventResponse.childtype
-                                childname = response.createdEventResponse.childname
-                                child = getattr(cb[0], childtype)[childname]
-                                cb[1](child)
-                            elif response.HasField("attributeChangedEventResponse"):
-                                value = response.attributeChangedEventResponse.value
-                                cb[1](_convert_variant_to_value(value))
-                            elif (
-                                response.HasField("modifiedEventResponse")
-                                or response.HasField("deletedEventResponse")
-                                or response.HasField("affectedEventResponse")
-                                or response.HasField(
-                                    "commandAttributeChangedEventResponse"
-                                )
-                            ):
-                                cb[1](cb[0])
-                            elif response.HasField("commandExecutedEventResponse"):
-                                command = response.commandExecutedEventResponse.command
-                                args = _convert_variant_to_value(
-                                    response.commandExecutedEventResponse.args
-                                )
-                                cb[1](cb[0], command, args)
+                    cb = self._cbs.get(response.tag, None)
```

Comment:
As tags are unique, we don't need to loop through the callback dictionary. Also, some callbacks can unsubscribe and delete other callbacks during which we cannot loop through the dictionary.